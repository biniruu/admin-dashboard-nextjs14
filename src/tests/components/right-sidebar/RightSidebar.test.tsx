import renderer from 'react-test-renderer'

import RightSidebar from '@components/right-sidebar/RightSidebar'

jest.mock('@components/right-sidebar/rightSidebarItems', () => {
  return [
    {
      notification: 'ðŸ”¥ available now',
      title: 'how to use the new version of the admin dashboard?',
      subtitle: 'takes 4 minutes to learn',
      desc: 'description',
      buttonIcon: <div>button icon</div>,
      button: 'watch',
      clickEvent: jest.fn(),
    },
    {
      notification: 'ðŸš€ coming soon',
      title: 'new server actions are available, partial pre-rendering is coming up!',
      subtitle: 'boost your productivity',
      desc: 'description',
      buttonIcon: <div>button icon</div>,
      button: 'learn',
      clickEvent: jest.fn(),
    },
  ]
})

describe('RightSidebar component', () => {
  test('should render the component', () => {
    const result = renderer.create(<RightSidebar />)

    expect(result).toMatchSnapshot()
  })
})
