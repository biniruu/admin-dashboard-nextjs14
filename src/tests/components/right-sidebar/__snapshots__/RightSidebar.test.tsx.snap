// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightSidebar component should render the component 1`] = `
<div
  className="fixed"
>
  <div
    className="relative mb-5 rounded-default px-6 py-5 item"
  >
    <div
      className="absolute bottom-0 right-0 size-1/2"
    >
      <img
        alt=""
        className="object-contain opacity-20"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        sizes="100vw"
        src="/_next/image?url=%2Fastronaut.png&w=3840&q=75"
        srcSet="/_next/image?url=%2Fastronaut.png&w=640&q=75 640w, /_next/image?url=%2Fastronaut.png&w=750&q=75 750w, /_next/image?url=%2Fastronaut.png&w=828&q=75 828w, /_next/image?url=%2Fastronaut.png&w=1080&q=75 1080w, /_next/image?url=%2Fastronaut.png&w=1200&q=75 1200w, /_next/image?url=%2Fastronaut.png&w=1920&q=75 1920w, /_next/image?url=%2Fastronaut.png&w=2048&q=75 2048w, /_next/image?url=%2Fastronaut.png&w=3840&q=75 3840w"
        style={
          {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="flex flex-col gap-6"
    >
      <div
        className="font-bold"
      >
        ðŸ”¥ available now
      </div>
      <h3
        className="title first-letter-capital"
      >
        how to use the new version of the admin dashboard?
      </h3>
      <div
        className="first-letter-capital text-xs font-medium text-text-soft"
      >
        takes 4 minutes to learn
      </div>
      <p
        className="text-sm text-text-soft"
      >
        description
      </p>
      <button
        className="flex w-max cursor-pointer items-center gap-default rounded-[0.3125rem] border-0 bg-[#5d57c9] p-default capitalize text-white"
        onClick={[Function]}
      >
        <div>
          button icon
        </div>
        watch
      </button>
    </div>
  </div>
  <div
    className="relative mb-5 rounded-default px-6 py-5 item"
  >
    <div
      className="flex flex-col gap-6"
    >
      <div
        className="font-bold"
      >
        ðŸš€ coming soon
      </div>
      <h3
        className="title first-letter-capital"
      >
        new server actions are available, partial pre-rendering is coming up!
      </h3>
      <div
        className="first-letter-capital text-xs font-medium text-text-soft"
      >
        boost your productivity
      </div>
      <p
        className="text-sm text-text-soft"
      >
        description
      </p>
      <button
        className="flex w-max cursor-pointer items-center gap-default rounded-[0.3125rem] border-0 bg-[#5d57c9] p-default capitalize text-white"
        onClick={[Function]}
      >
        <div>
          button icon
        </div>
        learn
      </button>
    </div>
  </div>
</div>
`;
